### DISCUSSION TITLE
"First dogecoin NFT" is not real

### DISCUSSION THREAD

**Discussion by GiverofMemory** [dogecoin/dogecoin#2750] [General]
Since this was censored by michi lumen on the issues discussion I will put this here to set the record straight.  News outlets are spreading the first dogecoin nft misinformation and it is contributing to the sort of hype that causes crashes since former and current dogecoin devs are perpetuating this.  Please read below to educate yourselves on NFT's and why smart contract platforms were created.

Was #2748
 
So what the person did is just reference an NFT on the dogecoin blockchain in the OP Return spot. There is no way to trade ownership of it. There are techniques, one called "colored coins" that can label coins to make tokens perhaps nft's that can be traded on standard blockchains like bitcoin or bitcoin cash and would also work with dogecoin. But if these worked well, they would be used. In fact tether originally launched on bitcoin but is now on ethereum for a reason. Dogecoin would need to be upgraded to a smart contract platform for token and nft trading to be a thing.

https://mobile.twitter.com/inevitable360/status/1470414541490110472?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1470414541490110472%7Ctwgr%5E%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Ffinbold.com%2Fdoge-developer-mints-first-ever-nft-on-the-dogecoin-blockchain%2F

https://chain.so/api/v2/tx/DOGE/19aeaa88859c04a333257f1119a77438ac08feec424c6ad3645a0679c8be9882

https://en.bitcoin.it/wiki/Colored_Coins

https://docs.google.com/document/d/1AnkP_cVZTCMLIzw4DvsW6M8Q2JC0lIzrTLuoWu2z1BE/edit#!

**Comment by qlpqlp** [comment]
Money is not real
Dogecoin is not real
Bitcoin is not real
Santa Claus and Christmas is not real
NFTs on Dogecoin Blockchain are not real

Its only real if you believe it is :)

**Reply by GiverofMemory** [reply]
If you want to be a real dogecoin dev how about you prove something as big as claim you minted a NFT on the dogecoin blockchain before trying to hype it on social media?  Do you not understand you are misleading hundreds of thousands of people that dogecoin is the next ethereum?  Learn about omnilayer and counterparty.io and colored coins and come up with a protocol, there is no way you can make what you did into a tradeable nft, you would need to encode more info into the transaction than the hash.  And even then it isn't an NFT like would be on a smart contract platform so isn't really comparable to get people hyped up about. Do the work before you go claiming crazy things and making developers and the community look bad

**Reply by Danny-dK** [reply]
Translated into normal enthusiastic not irritated language:
"_Hey man, I saw what you worked on in the past weeks. It's awesome that your interest in Dogecoin and blockchain has lead you to develop things for dogecoin that may be of use for the community at large. I do have some concerns though and would wonder if we could discuss this (although I fully realize that I am not being paid by you or paying you, so do not have much right to demanding things). I don't think that the Dogecoin NFT work you have done is technically an NFT minted on the dogecoin blockchain as one for example on the Ethereum blockchain (although I realize that you have never said that Dogecoin is similar to Ethereum now). I  don't fully understand if and how one would be able to trade ownership of the minted NFT on the dogecoin blockchain (it does not seem possible now?). Would you be willing to share your thoughts on whether this is true? If so, are there things that can be changed or developed to further enhance the Dogecoin experience (perhaps calling it something different than an NFT or topics that come to mind are omnilayer, colored coins, smart contracts). I have some experience / knowledge on this and would love to contribute in a positive way to the community at large (who I fully realize have their own responsibility)._"

**Reply by GiverofMemory** [reply]
Sure and unfortunately like was mentioned - crypto news is focused on hyping things and when they see real devs signing off on something they consider that a newsworthy event.  So we can all learn something from this, one of those things is that an NFT needs to be tradable and that tradable function requires a lot of work to achieve and can only be minted after that work is done.  After that if no one supports your new protocol like exchanges and wallets again it means nothing.  So the best way forward with this is to team up with a standard protocol like omni layer or counterparty.io to integrate dogecoin into the protocol.

Also imo when something like this happens more experienced devs should be quick to exhibit skepticism and try to tone down the hype before it gets out of control, instead of just ignoring it.

**Reply by qlpqlp** [reply]
My 2 cents!

Im a Boomer(42 years) and a DEV since 1998, i only started learning how Crypto really work in April 2021 and i still dont know nothing, learning everyday new things.

I saw that Dogecoin needed an hand to boost development, first i started to only make memes, than i was asked to join Million Doge Disco to make some memes, afther that i saw much more. Firstly i was only thinking on making some money for my familly, have still that on my mind, but i really wanted to help, so, with my "old methods codding skills" i started to play with the dogecoin nodes and try to learn how to talk with the Blockchain.

I see many things rong, but also many things right, and the "NFT" was pure luck because since 1998 until 2004 i worked on developping in VB programs to read TIFF images to be stored and add Cheksum to each one to be able to say its original and was not modified for legal propose, and wen i saw how Opensea/Rarible etc where saying NFT's i said to my self, thats not an NFT, really, you cannot prove you have the original.

Wen on day 13 i was making some tests to make something complitly diferent of the "NFT" discovery, i think to myself, why the hell no one is using to store as a real NFT? than made some testes and realize to myself, now this is an NFT, a Non Fungible Token that can prove the asset is original and was not modified.

If you can do it better, great, do it. I have all my code on GIT, you can use it, modify, do anything with it, will also share later my NFT's method on GIT to all.

**Reply by patricklodder** [reply]
> Also imo when something like this happens more experienced devs should be quick to exhibit skepticism and try to tone down the hype before it gets out of control, instead of just ignoring it.

Trying to put additional data on the blockchain is something that captivated lots of people for years and NFTs alone have become a multi-billion dollar industry this year. So the fact that people are learning and exploring is imho good, not bad. I understand your frustration with the press and hype, but this is not something we can change overnight. Influencing the influencers is a profession that is very different from being a blockchain developer, and developers becoming influencers overnight can be very very scary, but I think that same thing also goes for celebs and billionaires.

However, it is mostly a free world with some form of freedom of speech nearly everywhere, and Dogecoin is permissionless. So if in a year from now the only use case of Dogecoin is to store hashes of memes accompanied by dust-limit utxos, then so be it. I think it would be a suboptimal use of it **and** a suboptimal implementation of NFTs, but that's not something developers can prevent, or should aspire to; all that people can do is offer an alternative. Proposing the alternative takes time, so to the "quick" part, let's not. I also don't think any of the devs here has sufficient vision or experience to off-the-cuff voice all objections. The good news though is that the 24-hour news cycle has turned into a 2-hour one thanks to Twitter, so things are fairly transient and what is hyped today will be considered dead tomorrow. So let's not panic.

**Reply by GiverofMemory** [reply]
Thanks for the reply patrick.

It is unfortunate what happened and I feel a hair bit embarrassed because I don't want the "immature" prejudice against dogecoin to be perpetuated.  I think it is a little presumptuous probably for me to try to propose how this should have been handled, so I am sorry as coming off as "this is how it should have been handled" when I don't really know the best way.

I understand people are having fun and we want to promote that discovery process but also realize that anyone who is considered by the industry as a "dogecoin dev" that puts a lot of respect on them from the crypto industry no matter what their experience level, especially since the dogecoin dev team has done some great things lately like lowering the transaction fees (and getting listed on coinbase).  So people are respecting dogecoin devs and take what they say seriously, for better or worse.

**Reply by GiverofMemory** [reply]
So qlpqlp I applaud you for getting involved and jumping in and trying to do something.  But you also don't want to make false promises.  I know what you achieved here is perhaps being the first to list that image hash on dogecoin blockchain and you can sign for the address so you can prove you, qlpqlp, own that asset.  That is great.  But it isn't an NFT in the way that everyone understands it, and you probably got peoples hopes up (some probably expect NFT functionality like Ethereum in the near future now).

Like I referenced, "real" NFT's can be done on dogecoin or bitcoin or whatever, but the problem is it is not decentralized, you will have to make a protocol like Omni Layer did which is centralized and also you would have to make your own exchange and wallet to support your NFT protocol.  So until those things are well into development, I wouldn't try to say that there are NFT's on dogecoin.

**Reply by patricklodder** [reply]
> anyone who is considered by the industry as a "dogecoin dev" that puts a lot of respect on them from the crypto industry no matter what their experience level especially since the dogecoin dev team has done some great things lately like lowering the transaction fees (and getting listed on coinbase). So people are respecting dogecoin devs and take what they say seriously, for better or worse.

There's much more credit, both given and taken, than there is due, also to me personally. I expect that the fad will pass. People will get trusted for a pitch about something they didn't do and all I can hope is that they learn in time to not fail, for their own sake. Some will fail though, but others will succeed. 

A lot of the influence and fame for people in this community is founded on the principle of _fake it until you make it_. I haven't seen anyone make it yet and in fact I see a lot less hustling happen than one would expect, as all of it is still being in the fake-it stage. I'd be surprised if much will grow past that stage at all. People will just move on to fake other achievements and wishful thinking.

I wish everyone the best regardless, as this not something I think we should waste too much time on. Time spent there can not be spent on constructive things, so it will go at the cost of quality of this software and it would directly impact the quality of Dogecoin - which is already rather poor, imho.

_This week has already been pretty bad for me in terms of productivity due to all the distractions, so for the sake of progress, I'm going to switch to my zen laptop and do some work._ ❤️

**Reply by Doge369420** [reply]
Resident smooth brain chiming in here, I personally like the passion and creativity of inevitable 360.  Dogecoin is a bit silly and different, so not having a 'conventional' NFT is besides the point IMO.  I'd hate for someone working in this space to be too afraid of ridicule than to share some of their ideas.  Who knows, maybe a simplified record of ownership will become the new normal to help reduce blockchain congestion?  I don't think anyone simply sharing ideas can be responsible for any third party media hype. Please keep up the good work and creative juices flowing.

**Comment by Danny-dK** [comment]
> Since this was censored by michi lumen on the issues discussion 

@michilumin did not censor anything, she closed an issue in the issue section that should contain issues on dogecoin core. As she mentioned the take you present may have been better suited for social media (or in my view also appropriate to discuss in the discussion section here). In fact, you could've started a discussion with @qlpqlp right here, ::EDIT:: who just popped up in this thread, to discuss the thing he/she worked on and perhaps contribute on a better understanding or development. You asked in your original post https://github.com/dogecoin/dogecoin/issues/2748#issuecomment-995413784 why a dev would be 'hyping' it.... Michi has not been hyping it, from her posts you can tell she is interested in what is being developed. Devs can be interested and excited about what other people are doing. Devs are not responsible for the pisspoor job that the media is doing (in which actual good journalism is very difficult to find), nor are they responsible for people plunging into whatever crypto they get in without doing the bare minimum of research (i.e. this github, some basic google searches, social media, etc.). Chill and relax, enthusiasm for anything including dogecoin is always appreciated, just try not to turn enthusiasm into aggravation / irritation.

As a side note, I see the devs responding to posts / issues / discussion quite quickly and with calm composure, which I just see as incredible (especially considering the immense amount of people with malintent and that the devs have a main job aside from doing Dogecoin dev). Just saying: Thanks devs!

**Comment by antiquedogecoin** [comment]
<p class="p1">Hi! @antiquedoge here from <a href="http://dogeft.com"><span class="s1">dogeft.com</span></a>. We created the an nft showcase accepting Doge last year running from June-September before pausing to restructure a portion of the platform as it was seeing scalability issues. I started DogeFT bc I like Dogecoin art and wanted to collect but couldnt find dogecoin NFTs on opensea and bought a $3 eth NFT with an accidental $37 in gas and wtf and wanted to accept doge. Id like to start by saying that we built DogeFT with the goal of promoting functional adoption of Dogecoin and am excited to explore inevitable&#x2019;s solution and hope to adopt it in some form although our initial method utilizes a sidechain, but we also have a Doge only method. Im in support of community driven solutions to this challenge and am excited to support and contribute however helpful.</p>

<p class="p1">Our initial Dogecoin only approach was to carry just an image or text on a Dogecoin transaction but it doesnt carry ownership and are not technically NFTs as they arent unique tokens they are unique transactions. This has been live within VENFT App (at ve-nft.com) since last year from the Doge wallet whenever you send a transaction you can carry a message or text.<span class="Apple-converted-space">&nbsp;</span></p>

Here is an example: https://dogechain.info/tx/c0fd562e7cbfd3f32d5ed42301eaf553265e26b77ceada1e31393ca54c9e632f

This carries the message: https://gateway.ipfs.io/ipfs/Qmc9xS9a8TnWmU7AN4dtsbu4vU6hpEXpMNAeUdshFfg1wT

Dogecoin Transactional messages can be decoded by using an OP_RETURN decoder, such as opreturn.net: https://opreturn.net/c0fd562e7cbfd3f32d5ed42301eaf553265e26b77ceada1e31393ca54c9e632f

<p class="p1">I wanted to share info on our sidechain approach as well as its very similar to the GiverofMemory solution described above:<span class="Apple-converted-space">&nbsp;</span></p>

> <p class="p1">`but the problem is it is not decentralized, you will have to make a protocol like Omni Layer did which is centralized and also you would have to make your own exchange and wallet to support your NFT protocol. So until those things are well into development, I wouldn't try to say that there are NFT's on dogecoin.`</p>

<p class="p1">It's not just well into development, this platform will soon be operational after a successful beta last year.<span class="Apple-converted-space">&nbsp;</span></p>
<p class="p1">Here is a bit more on what&#x2019;s under the hood and our history:<span class="Apple-converted-space">&nbsp;</span></p>
<p class="p1">DogeFTs are NFTs minted using Dogecoin and proof of stake Neblio blockchain. Dogecoin is used as the currency layer, Neblio is used to carry the NFT data including Dogecoin info (side chain approach).</p>
<p class="p1">As we knew doge could carry limited data, we explored various alternative blockchains and had a few requirements:</p>
<ul class="ul1">
<li class="li1"><span class="s2"></span>Eco friendly</li>
<li class="li1"><span class="s2"></span>Low to no transaction processing costs</li>
<li class="li1"><span class="s2"></span>No pancakes required</li>
</ul>
<p class="p1">After doing a bit of homework we found Neblio blockchain (<a href="https://github.com/NeblioTeam/neblio"><span class="s1">https://github.com/NeblioTeam/neblio</span></a>) checked all the boxes. Cheap, fast, and easy to build with. To make things even better, there was already an open source Layer 2 project started which enables Neblio NFTs that had a web wallet that is a creators paradise and it already had a Doge wallet! </p>
<p class="p1">The Virtual Economy Framework (<a href="https://github.com/fyziktom/VirtualEconomyFramework"><span class="s1">https://github.com/fyziktom/VirtualEconomyFramework</span></a>) supports virtually any type of NFT from images on ipfs, event tickets, music, invoices, products, and you can basically build any NFTs types you can imagine. I connected with @fyziktom who is the original author of the framework and we started building!</p> 
<p class="p1">Our initial solution mints basically all of the relevant NFT details (author, ownership, and original content) and Dogecoin details on the Neblio blockchain.</p>
<p class="p1">So lets check one out! Here is an early days DogeFT:</p>
<p class="p3"><a href="https://explorer.nebl.io/tx/631858d029b89ad78b2436d7fdbf94f75d199c9d87207ebcaf896e11857cad64">https://explorer.nebl.io/tx/3511f6915afafd4f21ce3512efb060bfb4cb4f5c0ee4e761430575be138d7447</a></p>
<p class="p1">This token carries the following metadata:</p>


Key | Value
-- | --
NFT | true
Type | NFT Image
Name | Doge Diptych 3
Author | @antiquedoge
Description | Doge Diptych 3 NFT
Image | https://gateway.ipfs.io/ipfs/QmQFHoCaZrjVs4Rz1b8w7czfxegUVkBUMUzh4jWSm2QvUZ
Link |  
DogePrice | 100
DogeAddress | DJkjBPyBB58xUvqdxzJvSsd9HnNU7wsyqX
SourceUtxo | 73cdbb080e6106f84744a9e0d483071cb8432a18900738b79e2f1dc3dccb20ab
NFT FirstTx | true

<p class="p1">When an author submits their NFTs to the marketplace a transaction is created with the metadata including the Dogecoin Address and NFT details.<span class="Apple-converted-space">&nbsp;</span></p>
<p class="p1">We created multiple reward schemes to enable different contract settings, for example to donate to a different charity we loaded a new reward scheme.</p>
<p class="p1">Here is the NFT submission to the showcase:</p>
<p class="p3"><a href="https://explorer.nebl.io/tx/478def2467e212c3f1640266ba5e15bd87b47080f2ffcdc92b5a0b66ae390e2c">https://explorer.nebl.io/tx/478def2467e212c3f1640266ba5e15bd87b47080f2ffcdc92b5a0b66ae390e2c</a></p>

Key | Value
-- | --
NFT | true
Type | NFT Image
Name | Doge Diptych 3
Author | @antiquedoge
Description | Doge Diptych 3 NFT
Image | https://gateway.ipfs.io/ipfs/QmQFHoCaZrjVs4Rz1b8w7czfxegUVkBUMUzh4jWSm2QvUZ
Link |  
DogePrice | 100
DogeAddress | DJkjBPyBB58xUvqdxzJvSsd9HnNU7wsyqX
DogeftInfo | {"License":"","Coppies":0,"AuthorUrl":"","RewardSchemeName":"DogeftMain","AuthorDogeAddress":"DJkjBPyBB58xUvqdxzJvSsd9HnNU7wsyqX"}
SourceUtxo | 3511f6915afafd4f21ce3512efb060bfb4cb4f5c0ee4e761430575be138d7447

<p class="p1">When an NFT is sold, the contract calls the Dogeftinfo from the nft to reward based on the schema. So for example they can be loaded with a % payout to artist, and % to charity.<span class="Apple-converted-space">&nbsp; </span>In 2021 we donated 1073 Doge to muttville senior dog rescue and 744.5 Doge to American cancer.<span class="Apple-converted-space">&nbsp;</span></p>

<p class="p1">One of the more difficult challenges with this approach is centralization. We are working on building the marketplace functionality directly into VENFT app to help decentralize the solution further and hopefully unlock a really cool ecosystem. More to come there!<span class="Apple-converted-space">&nbsp;</span></p>

<p class="p1">If more info on our project would be helpful in any way please dont hesitate to ping myself or @fyziktom.<span class="Apple-converted-space">&nbsp; </span>Looking forward to sharing updates on our progress and continuing to expand the Dogecoin arts community together.<span class="Apple-converted-space">&nbsp;</span></p>

</span>Thanks for the support doge devs. Were rooting for you! To the moon!<span class="Apple-converted-space">&nbsp;</span></p>
</body>
</html>


Update: We have been exploring our Dogecoin only solution a good bit and have come up with an alternative approach. Effectively it is accomplished by carrying the ownership data as a JSON on IPFS & storing the CID (IPFS content identifier) on OP_RETURN. Here is a prototype we are calling NFÐs. More to come on this solution.

```
"OldOwnerAddress":"OLD_OWNER_ADDRESS",
"NewOwnerAddress":"NEW_OWNER_ADDRESS",
"Utxo":"",
"UtxoIndex":,
"Name":"NAME",
"AuthorName":"AUTHOR",
"AuthorAddress":"AUTHOR_ADDRESS",
"ImageLink":"IPFS LINK,
"Price":PRICE,
"Time":"TIME"
"Reward":"{REWARD_SCHEMA}"
```
Here is an prototype transaction carrying ownership data on IPFS
https://dogechain.info/tx/edea36aab42aaace9ae5055c2418bec7460dc947263ede945bf2e0d7687b229f

Proof: https://opreturn.net/edea36aab42aaace9ae5055c2418bec7460dc947263ede945bf2e0d7687b229f

Tx carries message (CID): 
"QmRPSe9N1SBVo3R6CpiqQ7GyiguJ69fSEprJWh9pSc8Vmk"

IPFS link: https://gateway.ipfs.io/ipfs/QmRPSe9N1SBVo3R6CpiqQ7GyiguJ69fSEprJWh9pSc8Vmk

**Reply by qlpqlp** [reply]
Great work you have done :)

Continue to develop and don't give up :)

Anyone can build anything its Opensource, so keep building things :)

Dogecoin needs more DEVs more utility and more ideas  ;)

**Reply by antiquedogecoin** [reply]
Thanks inevitable for the words of support! We are taking Dogecoin to some new and unique places together and I CANT WAIT!

Much excited!

We will be busy when Dogecoin is #1! 🌕

**Comment by ShibeFaceSkrill** [comment]
> if these worked well, they would be used.

yes but they are used :eyes: I just bought an NFT from [Dogeparty](https://www.dogeparty.net/), (counterparty on dogecoin).

and yes, [Counterparty](https://counterparty.io/) is still alive and well along with, the original NFT on Bitcoin blockchain, [RarePepe](http://rarepepedirectory.com/) and many other tokens.

It's not widely used on Bitcoin because of Bitcoin Politics (but secures high-value assets, nonetheless)

it hasn't been used widely on Dogecoin because NFT have only become popular in the last couple years.

Dogeparty certainly won't be the last attempt to use dogecoin for NFT, but it's quite servicable and will grow nicely into the coming bull-season.

Check it out: https://github.com/DogepartyXDP


that's the funniest thing about this is the title is accurate, but for the opposite reason of why you give.
